sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/PrivateFieldUsedAsLocalVariable.cs

```diff
@@ -267,6 +267,7 @@ private static bool IsWrite(SyntaxNodeWithSymbol<SyntaxNode, ISymbol> fieldRefer
 
                 bool IsLeftSideOfAssignment(SyntaxNode syntaxNode) =>
                     syntaxNode.Parent is AssignmentExpressionSyntax assignmentExpression &&
+                    syntaxNode.Parent.Kind() == SyntaxKind.SimpleAssignmentExpression &&

```

Comment:
This could be `IsKind(SyntaxKind.SimpleAssignmentExpression) and could be before the cast above...